### PR TITLE
feat: support OTEL autoexport

### PR DESCRIPTION
**Pull request by shanduur**
## Description

This pull request includes the following changes:
- Updated tooling to ensure compatibility and improved development workflow.
- Added support for OpenTelemetry (OTel) auto instrumentation, enhancing observability.
- Dropped support for OpenTelemetry (OTel) sidecar, simplifying the setup.
- Bumped application version in the Helm chart for deployment.
- Fixed tracing shutdown issue, ensuring proper termination of tracing processes.
- Fixed the auto-exporter for Prometheus, enabling correct metric exports.
- Enabled Prometheus exporter as default and added pod annotations for better monitoring and metrics collection.

### CODE REVIEW

File: pkg/observability/tracing/tracing.go [hunk 1]
```diff
@@ -56,7 +46,7 @@ func Setup(ctx context.Context, resource *resource.Resource, protocol string) (_
 func registerTraceExporter(res *resource.Resource, exporter sdktrace.SpanExporter) (func(context.Context) error, error) {
 	options := []sdktrace.TracerProviderOption{
 		sdktrace.WithBatcher(exporter),
-		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		sdktrace.WithSampler(sdktrace.TraceIDRatioBased(defaultSamplingRatio)),
```
Comment (nesv) [inline pkg/observability/tracing/tracing.go:49]: Is there any way for the user to adjust this sampling rate?

**Review by nesv — APPROVED** [review]

File: pkg/observability/tracing/tracing.go (hunk 1 above)
Comment (shanduur) [reply pkg/observability/tracing/tracing.go]: Good catch! I fixed it, so now `OTEL_TRACES_SAMPLER` and `OTEL_TRACES_SAMPLER_ARG` can be set together, so the sampler is configured dynamically, e.g.:

```sh
OTEL_TRACES_SAMPLER=traceidratio
OTEL_TRACES_SAMPLER_ARG=0.001
```

Comment (shanduur) [reply pkg/observability/tracing/tracing.go]: (Defaults to [`parentbased_always_on`](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#:~:text=Propagator%20only%20once.-,OTEL_TRACES_SAMPLER,-Sampler%20to%20be).

Comment (nesv) [reply pkg/observability/tracing/tracing.go]: Sounds good!

**Review by nesv — APPROVED** [review]

**Pull request merged**